### ISSUE TITLE
Verify JSON parsing for AWSTimestamp with TZ offset in seconds

### ISSUE THREAD

**Issue by palpatim** [aws-amplify/amplify-ios#286]
Cloned from https://github.com/aws-amplify/amplify-android/issues/221:

The issue below is obviously Android specific, but we need to validate that our code supports this case, since AppSync handles this ISO8601 extension.

Options per @raphkim:

1. truncate the extended precision before parsing
2. find non-iso8601 restricted date-time package
3. write parsing logic from scratch

Be sure to coordinate with Android & JS to make sure all platforms handle it the same.

---

`AWSDate` supports seconds field in the timezone offset, which is not supported by ISO8601 format. This causes Gson to fail to parse date such as "1970-01-01-01:00:00". 

Sample code to reproduce:
```
Gson gson = new Gson();
gson.fromJson("\"1970-01-01-01:00:00\"", Date.class);
```

logcat report from integration test (breaks CircleCI workflow):
```
    --------- beginning of crash
2020-01-06 16:59:13.548 31371-31406/com.amplifyframework.api.aws.test E/AndroidRuntime: FATAL EXCEPTION: OkHttp Dispatcher
    Process: com.amplifyframework.api.aws.test, PID: 31371
    com.google.gson.JsonSyntaxException: 1970-01-01-07:00:39
        at com.google.gson.internal.bind.DateTypeAdapter.deserializeToDate(DateTypeAdapter.java:87)
        at com.google.gson.internal.bind.DateTypeAdapter.read(DateTypeAdapter.java:75)
        at com.google.gson.internal.bind.DateTypeAdapter.read(DateTypeAdapter.java:46)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
        at com.google.gson.Gson.fromJson(Gson.java:932)
        at com.google.gson.Gson.fromJson(Gson.java:1003)
        at com.google.gson.Gson.fromJson(Gson.java:975)
        at com.amplifyframework.api.aws.GsonGraphQLResponseFactory.parseDataAsList(GsonGraphQLResponseFactory.java:198)
        at com.amplifyframework.api.aws.GsonGraphQLResponseFactory.buildSingleArrayResponse(GsonGraphQLResponseFactory.java:132)
        at com.amplifyframework.api.graphql.GraphQLOperation.wrapMultiResultResponse(GraphQLOperation.java:71)
        at com.amplifyframework.api.aws.SingleArrayResultOperation.access$100(SingleArrayResultOperation.java:47)
        at com.amplifyframework.api.aws.SingleArrayResultOperation$OkHttpCallback.onResponse(SingleArrayResultOperation.java:140)
        at okhttp3.RealCall$AsyncCall.run(RealCall.kt:138)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:764)
     Caused by: java.text.ParseException: Failed to parse date ["1970-01-01-07:00:39"]: Mismatching time zone indicator: GMT-07:00:39 given, resolves to GMT
        at com.google.gson.internal.bind.util.ISO8601Utils.parse(ISO8601Utils.java:274)
        at com.google.gson.internal.bind.DateTypeAdapter.deserializeToDate(DateTypeAdapter.java:85)
        at com.google.gson.internal.bind.DateTypeAdapter.read(DateTypeAdapter.java:75) 
        at com.google.gson.internal.bind.DateTypeAdapter.read(DateTypeAdapter.java:46) 
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131) 
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222) 
        at com.google.gson.Gson.fromJson(Gson.java:932) 
        at com.google.gson.Gson.fromJson(Gson.java:1003) 
        at com.google.gson.Gson.fromJson(Gson.java:975) 
        at com.amplifyframework.api.aws.GsonGraphQLResponseFactory.parseDataAsList(GsonGraphQLResponseFactory.java:198) 
        at com.amplifyframework.api.aws.GsonGraphQLResponseFactory.buildSingleArrayResponse(GsonGraphQLResponseFactory.java:132) 
        at com.amplifyframework.api.graphql.GraphQLOperation.wrapMultiResultResponse(GraphQLOperation.java:71) 
        at com.amplifyframework.api.aws.SingleArrayResultOperation.access$100(SingleArrayResultOperation.java:47) 
        at com.amplifyframework.api.aws.SingleArrayResultOperation$OkHttpCallback.onResponse(SingleArrayResultOperation.java:140) 
        at okhttp3.RealCall$AsyncCall.run(RealCall.kt:138) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764) 
     Caused by: java.lang.IndexOutOfBoundsException: Mismatching time zone indicator: GMT-07:00:39 given, resolves to GMT
        at com.google.gson.internal.bind.util.ISO8601Utils.parse(ISO8601Utils.java:240)
        at com.google.gson.internal.bind.DateTypeAdapter.deserializeToDate(DateTypeAdapter.java:85) 
        at com.google.gson.internal.bind.DateTypeAdapter.read(DateTypeAdapter.java:75) 
        at com.google.gson.internal.bind.DateTypeAdapter.read(DateTypeAdapter.java:46) 
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131) 
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222) 
        at com.google.gson.Gson.fromJson(Gson.java:932) 
        at com.google.gson.Gson.fromJson(Gson.java:1003) 
        at com.google.gson.Gson.fromJson(Gson.java:975) 
        at com.amplifyframework.api.aws.GsonGraphQLResponseFactory.parseDataAsList(GsonGraphQLResponseFactory.java:198) 
        at com.amplifyframework.api.aws.GsonGraphQLResponseFactory.buildSingleArrayResponse(GsonGraphQLResponseFactory.java:132) 
        at com.amplifyframework.api.graphql.GraphQLOperation.wrapMultiResultResponse(GraphQLOperation.java:71) 
        at com.amplifyframework.api.aws.SingleArrayResultOperation.access$100(SingleArrayResultOperation.java:47) 
        at com.amplifyframework.api.aws.SingleArrayResultOperation$OkHttpCallback.onResponse(SingleArrayResultOperation.java:140) 
        at okhttp3.RealCall$AsyncCall.run(RealCall.kt:138) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764) 
```

**Comment by palpatim** [comment]
More general statement: All platforms will support AWS AppSync scalar types as specified in https://docs.aws.amazon.com/appsync/latest/devguide/scalars.html

For each of the scalar types, we will have a set of shared test case definitions such that a given input string defined as a specific AppSync scalar type will serialize to a platform-specific representation that can be verified. E.g.:
> `2020-01-09T01:23:45.678+00:00:00` as an AWSDateTime returns an object whose representation in nanoseconds since the Unix epoch is `1234567`

**Comment by drochetti** [comment]
fixed by #299
